************* Module app.auth
app/auth.py:145:8: W0707: Consider explicitly re-raising using 'except JWTError as exc' and 'raise credentials_exception from exc' (raise-missing-from)
app/auth.py:8:0: C0411: standard import "datetime.datetime" should be placed before third party imports "fastapi.APIRouter", "fastapi.security.OAuth2PasswordBearer", "sqlalchemy.orm.Session", "passlib.context.CryptContext", "jose.JWTError" (wrong-import-order)
app/auth.py:3:0: W0611: Unused status imported from fastapi (unused-import)
************* Module app.database
app/database.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/database.py:9:0: C0116: Missing function or method docstring (missing-function-docstring)
************* Module app.main
app/main.py:72:0: C0301: Line too long (102/100) (line-too-long)
app/main.py:116:0: C0301: Line too long (111/100) (line-too-long)
app/main.py:124:0: C0301: Line too long (102/100) (line-too-long)
app/main.py:136:0: C0301: Line too long (105/100) (line-too-long)
app/main.py:205:0: C0305: Trailing newlines (trailing-newlines)
app/main.py:8:0: R0402: Use 'from app import models' instead (consider-using-from-import)
app/main.py:9:0: R0402: Use 'from app import database' instead (consider-using-from-import)
app/main.py:17:0: W0404: Reimport 'HTTPException' (imported line 3) (reimported)
app/main.py:17:0: C0413: Import "from fastapi import HTTPException" should be placed at the top of the module (wrong-import-position)
app/main.py:22:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:26:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:37:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:68:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:72:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:83:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:116:63: E1102: func.count is not callable (not-callable)
app/main.py:119:18: E1102: func.count is not callable (not-callable)
app/main.py:129:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:136:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:147:0: C0116: Missing function or method docstring (missing-function-docstring)
app/main.py:17:0: C0411: third party import "fastapi.HTTPException" should be placed before first party imports "app.auth.get_current_user", "app.models", "app.database"  (wrong-import-order)
app/main.py:17:0: C0412: Imports from package fastapi are not grouped (ungrouped-imports)
************* Module app.models
app/models.py:9:0: R0903: Too few public methods (0/2) (too-few-public-methods)
app/models.py:30:0: R0903: Too few public methods (0/2) (too-few-public-methods)
app/models.py:49:0: R0903: Too few public methods (0/2) (too-few-public-methods)
************* Module tests.main_test
tests/main_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
tests/main_test.py:11:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:12:4: W0621: Redefining name 'db' from outer scope (line 11) (redefined-outer-name)
tests/main_test.py:16:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:16:22: W0621: Redefining name 'db' from outer scope (line 11) (redefined-outer-name)
tests/main_test.py:16:22: W0613: Unused argument 'db' (unused-argument)
tests/main_test.py:23:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:23:20: W0621: Redefining name 'db' from outer scope (line 11) (redefined-outer-name)
tests/main_test.py:41:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:41:22: W0621: Redefining name 'db' from outer scope (line 11) (redefined-outer-name)
tests/main_test.py:41:22: W0613: Unused argument 'db' (unused-argument)
tests/main_test.py:47:0: C0413: Import "from app.auth import create_access_token" should be placed at the top of the module (wrong-import-position)
tests/main_test.py:49:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:49:26: W0621: Redefining name 'db' from outer scope (line 11) (redefined-outer-name)
tests/main_test.py:75:0: C0116: Missing function or method docstring (missing-function-docstring)
tests/main_test.py:5:0: C0411: third party import "sqlalchemy.orm.Session" should be placed before first party imports "app.main.app", "app.models.User", "app.database.SessionLocal"  (wrong-import-order)
tests/main_test.py:6:0: C0411: third party import "pytest" should be placed before first party imports "app.main.app", "app.models.User", "app.database.SessionLocal"  (wrong-import-order)
tests/main_test.py:47:0: C0412: Imports from package app are not grouped (ungrouped-imports)
tests/main_test.py:4:0: W0611: Unused engine imported from app.database (unused-import)
tests/main_test.py:5:0: W0611: Unused Session imported from sqlalchemy.orm (unused-import)

------------------------------------------------------------------
Your code has been rated at 7.55/10 (previous run: 7.55/10, +0.00)

